applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.cpp

```diff
@@ -359,4 +382,353 @@ void GeometryUtilities::CalculateUnitNormals()
     }
 }
 
+void GeometryUtilities::CalculateGaussianCurvature() {
+
+    KRATOS_TRY
+
+    // check if surface normals are necessary (only for Taubin method)
+    bool normals_necessary = false;
+    for (const auto& rNode : mrModelPart.Nodes()) {
+        std::string technique = this->GetCurvatureTechnique(rNode);
+        if (technique == "Taubin") {
+            normals_necessary = true;
+            break;
+        }
+    }
+    if (normals_necessary) this->ComputeUnitSurfaceNormals();
+
+    // check if edge modelpart is necessary (only for Meyer method)
+    bool edge_necessary = false;
+    for (const auto& rNode : mrModelPart.Nodes()) {
+        std::string technique = this->GetCurvatureTechnique(rNode);
+        if (technique == "Meyer") {
+            edge_necessary = true;
+            break;
+        }
+    }
+    if (edge_necessary) {
+        ModelPart& r_edge_model_part = mrModelPart.HasSubModelPart(mrModelPart.Name() + "_edges") ? mrModelPart.GetSubModelPart(mrModelPart.Name() + "_edges") : mrModelPart.CreateSubModelPart(mrModelPart.Name() + "_edges");
+        if (r_edge_model_part.Nodes().size() == 0) {
+            this->ExtractEdgeNodes(mrModelPart.Name() + "_edges");
+        }
+    }
+
+    block_for_each(mrModelPart.Nodes(), [&](NodeType &rNode) {
+
+        const auto& r_condition_neighbours = rNode.GetValue(NEIGHBOUR_CONDITIONS).GetContainer();
+        double& r_gaussian_curvature = rNode.FastGetSolutionStepValue(GAUSSIAN_CURVATURE);
+
+        std::string technique = this->GetCurvatureTechnique(rNode);
+
+        // For quadratic elements (e.g. 6NTriangle)
+        if (technique == "curvature_tensor") {
```

Comment:
Can you add 3 functions: ComputeCurvatureTensorforNode(), ComputeMeyerforNode(), ComputeTaubinForNode() and replace all part of codes inside the if statements?? It should improve the code and add modularity. At least the computation part itself.

Reply:
done